.github/actions/secure-project-checkout-go/action.yml

```
@@ -44,19 +44,28 @@ runs:
         path: ${{ inputs.path }}
 
     - name: Validate Go version.
+      id: validate
       shell: bash
       env:
         GO_VERSION: ${{ inputs.go-version }}
         GO_VERSION_FILE: ${{ inputs.go-version-file }}
+        INPUT_PATH: ${{ inputs.path }}
       run: |
         set -euo pipefail
         if [ "${GO_VERSION}" == "" ] && [ "${GO_VERSION_FILE}" == "" ]; then
             echo "::error::One of go-version or go-version-file should be specified."
             exit 1
         fi
+        if [ "${GO_VERSION}" != "" ]; then
+          echo "go_version=${GO_VERSION}" >> "$GITHUB_OUTPUT"
+        fi
+        if [ "${GO_VERSION_FILE}" != "" ]; then
+          # NOTE: We use realpath to remove the trailing '/' if present.
+          echo "go_version_file=$(realpath ${INPUT_PATH})/${GO_VERSION_FILE}" >> "$GITHUB_OUTPUT"
```

TODO: need to tests this. I'm not sure if this file can be absolute or needs to be relative the GITHUB_WORKSPACE

---

https://github.com/actions/setup-go/blob/main/src/main.ts#L153 it uses `fs.existsSync` which understands absolute paths.

---

Yah, it's either absolute or relative to `GITHUB_WORKSPACE` afaik

This would also work to remove at least one right trailing `/`
```suggestion
          echo "go_version_file=${INPUT_PATH%%/}/${GO_VERSION_FILE}" >> "$GITHUB_OUTPUT"
```